### PR TITLE
Add mypy

### PR DESCRIPTION
**Pull request by Natim**
Based on #8 

Please merge #7 and #8 first and then I will rebase this one.

### CODE REVIEW

File: alma/endpoints/orders.py [hunk 1]
```diff
@@ -13,10 +13,10 @@ def update(self, order_id, data):
         return Order(response.json)
 
     def fetch_all(self, limit: int = 20, starting_after: str = None, **filters):
-        args = {"limit": limit}
+        args = {"limit": str(limit)}
 
         if starting_after:
-            args["starting_after"] = starting_after
+            args["starting_after"] = str(starting_after)
```
Comment (olance) [inline alma/endpoints/orders.py]: Not sure I get the need for that change? `starting_after` is already annotated as a `str`, and cannot be `None` in that code branch

File: alma/endpoints/payments.py [hunk 2]
```diff
@@ -19,9 +19,9 @@ def create(self, data):
         return Payment(response.json)
 
     def fetch_all(self, limit: int = 20, states: list = None, starting_after: str = None):
-        args = {"limit": limit}
+        args = {"limit": str(limit)}
         if starting_after:
-            args["starting_after"] = starting_after
+            args["starting_after"] = str(starting_after)
```
Comment (olance) [inline alma/endpoints/payments.py]: Same as previous comment

File: pyproject.toml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [tool.black]
-line-length = 110
+line-length = 99
```
Comment (olance) [inline pyproject.toml]: Please undo this, unless you've got a great argument for what looks like another arbitrary value?

```suggestion
line-length = 110
```

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  - repo: https://github.com/pre-commit/mirrors-isort
+    rev: v4.3.21
+    hooks:
+      - id: isort
+
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 3.7.9
+    hooks:
+      - id: flake8
+        args: [--max-line-length=99]
```
Comment (olance) [inline .pre-commit-config.yaml:30]:
```suggestion
        args: [--max-line-length=110]
```

File: setup.py [hunk 5]
```diff
@@ -10,7 +11,7 @@
 
 # This call to setup() does all the work
 setup(
-    name="alma-python-client",
+    name="alma-client",
```
Comment (olance) [inline setup.py]: 👍

File: tox.ini [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+deps = -rdev-requirements.txt
+install_command = pip install {opts} {packages}
+
+[testenv:flake8]
+commands = flake8 alma
+deps =
+    flake8
+
+[flake8]
+max-line-length = 99
```
Comment (olance) [inline tox.ini]:
```suggestion
max-line-length = 110
```

File: alma/endpoints/orders.py (hunk 1 above)
Comment (Natim) [reply alma/endpoints/orders.py]: Good catch!, I guess mypy was puzzled to have limit as an int and starting_after as a str.

**Pull request merged**